dashboard/src/components/AppView/AppControls/UpgradeButton/UpgradeButton.tsx

```diff
@@ -9,16 +10,18 @@ interface IUpgradeButtonProps {
   namespace: string;
   releaseName: string;
   releaseStatus: InstalledPackageStatus | undefined | null;
+  plugin: Plugin;
 }
 
 export default function UpgradeButton({
   cluster,
   namespace,
   releaseName,
   releaseStatus,
+  plugin,
 }: IUpgradeButtonProps) {
   return (
-    <Link to={url.app.apps.upgrade(cluster, namespace, releaseName)}>
+    <Link to={url.app.apps.upgrade(cluster, namespace, releaseName, plugin)}>
```

Comment:
Similar point regarding uniqueness (identifier instead of releaseName), and whether it's worth thinking about passing an `InstalledPackageRef` instead of many fields.

Reply:
Done!